Dedicated server NullReferenceException

Got this exception in my logs, server crashed.

```
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Network.Session+<PingFromClient>c__AnonStorey2.<>m__0 (OpenRA.Network.ClientPing p) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.Single[ClientPing] (IEnumerable`1 source, System.Func`2 predicate, Fallback fallback) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.SingleOrDefault[ClientPing] (IEnumerable`1 source, System.Func`2 predicate) [0x00000] in <filename unknown>:0 
  at OpenRA.Network.Session.PingFromClient (OpenRA.Network.Client client) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrder (OpenRA.Server.Connection conn, OpenRA.Server.ServerOrder so) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrders (OpenRA.Server.Connection conn, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.DispatchOrders (OpenRA.Server.Connection conn, Int32 frame, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadData (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.<Server>m__0 (System.Object _) [0x00000] in <filename unknown>:0 
  at System.Threading.Thread.StartInternal () [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Network.Session+<PingFromClient>c__AnonStorey2.<>m__0 (OpenRA.Network.ClientPing p) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.Single[ClientPing] (IEnumerable`1 source, System.Func`2 predicate, Fallback fallback) [0x00000] in <filename unknown>:0 
  at System.Linq.Enumerable.SingleOrDefault[ClientPing] (IEnumerable`1 source, System.Func`2 predicate) [0x00000] in <filename unknown>:0 
  at OpenRA.Network.Session.PingFromClient (OpenRA.Network.Client client) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrder (OpenRA.Server.Connection conn, OpenRA.Server.ServerOrder so) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.InterpretServerOrders (OpenRA.Server.Connection conn, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.DispatchOrders (OpenRA.Server.Connection conn, Int32 frame, System.Byte[] data) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Connection.ReadData (OpenRA.Server.Server server) [0x00000] in <filename unknown>:0 
  at OpenRA.Server.Server.<Server>m__0 (System.Object _) [0x00000] in <filename unknown>:0 
  at System.Threading.Thread.StartInternal () [0x00000] in <filename unknown>:0 
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8785468-dedicated-server-nullreferenceexception?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/OpenRA.Game/Server/Server.cs
+++ b/OpenRA.Game/Server/Server.cs
@@ -489,7 +489,11 @@ namespace OpenRA.Server
 							break;
 						}
 
-						var pingFromClient = LobbyInfo.PingFromClient(GetClient(conn));
+						var client = GetClient(conn);
+						if (client == null)
+							return;
+
+						var pingFromClient = LobbyInfo.PingFromClient(client);
 						if (pingFromClient == null)
 							return;
 

--- a/OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs
+++ b/OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs
@@ -186,8 +186,11 @@ namespace OpenRA.Mods.Common.Server
 								server.LobbyInfo.Clients.Remove(occupant);
 								server.SyncLobbyClients();
 								var ping = server.LobbyInfo.PingFromClient(occupant);
-								server.LobbyInfo.ClientPings.Remove(ping);
-								server.SyncClientPing();
+								if (ping != null)
+								{
+									server.LobbyInfo.ClientPings.Remove(ping);
+									server.SyncClientPing();
+								}
 							}
 							else
 							{
@@ -221,7 +224,11 @@ namespace OpenRA.Mods.Common.Server
 						{
 							server.LobbyInfo.Clients.Remove(occupant);
 							var ping = server.LobbyInfo.PingFromClient(occupant);
-							server.LobbyInfo.ClientPings.Remove(ping);
+							if (ping != null)
+							{
+								server.LobbyInfo.ClientPings.Remove(ping);
+								server.SyncClientPing();
+							}
 						}
 
 						server.SyncLobbyClients();

